src/sql/base/browser/ui/taskbar/collapsibleActionbar.ts

```diff
@@ -0,0 +1,294 @@
+			}
+		})));
+
+		this._register(DOM.addDisposableListener(this._overflow, DOM.EventType.KEY_DOWN, (e: KeyboardEvent) => {
+			let event = new StandardKeyboardEvent(e);
+
+			// Close overflow if Escape is pressed
+			if (event.equals(KeyCode.Escape)) {
+				this.hideOverflowDisplay();
+				this._moreActionsElement.focus();
+			} else if (event.equals(KeyCode.UpArrow) && this._focusedItem !== this._actionsList.childNodes.length - 1) { // up arrow on first element in overflow should not move out from the overflow
+				this.focusPrevious();
+			} else if (event.equals(KeyCode.DownArrow) && this._focusedItem !== this._actionsList.childNodes.length + this._overflow.childNodes.length - 2) { // down arrow on last element shouldn't move out from overflow
+				this.focusNext();
+			} else if (event.equals(KeyMod.Shift | KeyCode.Tab)) {
+				this._moreActionsElement.focus();
+			} else if (event.equals(KeyCode.Tab)) {
+				this.hideOverflowDisplay();
+			}
+			DOM.EventHelper.stop(event, true);
+		}));
+
+		this._moreItemElement.appendChild(this._moreActionsElement);
+		this._actionsList.appendChild(this._moreItemElement);
+		this._items.push(undefined); // add place holder for more item element
+	}
+
+	private hideOverflowDisplay(): void {
+		this._overflow.style.display = 'none';
+	}
+
+	private toggleOverflowDisplay(): void {
+		this._overflow.style.display = this._overflow.style.display === 'block' ? 'none' : 'block';
+	}
+
+	protected updateFocusedItem(): void {
+		let actionIndex = 0;
+		for (let i = 0; i < this._actionsList.children.length; i++) {
+			let elem = this._actionsList.children[i];
+
+			if (DOM.isAncestor(document.activeElement, elem)) {
+				this._focusedItem = actionIndex;
+				break;
+			}
+
+			if (elem.classList.contains('action-item') && i !== this._actionsList.children.length - 1) {
+				actionIndex++;
+			}
+		}
+
+		// move focus to overflow items if there are any
+		if (this._overflow) {
+			for (let i = 0; i < this._overflow.children.length; i++) {
+				let elem = this._overflow.children[i];
+
+				if (DOM.isAncestor(document.activeElement, elem)) {
+					this._focusedItem = actionIndex;
+					break;
+				}
+
+				if (elem.classList.contains('action-item')) {
+					actionIndex++;
+				}
+			}
+		}
+	}
+
+	/**
+	 * Push an HTML Element onto the action bar UI in the position specified by options.
+	 * Pushes to the last position if no options are provided.
+	 */
+	public pushElement(element: HTMLElement, options: IActionOptions = {}): void {
+		super.pushElement(element, options);
+		this.resizeToolbar();
+	}
+
+	/**
+	 * Push an action onto the action bar UI in the position specified by options.
+	 * Pushes to the last position if no options are provided.
+	 */
+	public pushAction(arg: IAction | IAction[], options: IActionOptions = {}): void {
+		super.pushAction(arg, options);
+		this.resizeToolbar();
+	}
+
+	protected focusNext(): void {
+		if (typeof this._focusedItem === 'undefined') {
```

Comment:
Do you need to re-implement this here? I didn't look too closely but it seems to be mostly the same as the base ActionBar implementation (same for other similar functions)

Reply:
For the focus ones, they need to be reimplemented to skip the placeholder overflow item. The other functions that are re-implemented in this class also need to be there to handle the showing the overflow or for resizing.